lib/engine/step/g_18_al/assign.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require_relative '../assign'
+
+module Engine
+  module Step
+    module G18AL
+      class Assign < Assign
+        def process_assign(action)
+          company = action.entity
+          raise GameError, "#{company.owner.name} owns no trains" if company.owner.trains.empty?
+
+          target = action.target
+          hexes = company.abilities(:assign_hexes)&.hexes
+          reachable = @game.graph.reachable_hexes(company.owner).find { |h, _| h.id == target.id }
```

Comment:
only run this unless @game.is_loading because this is very expensive

check for other examples

Reply:
Ok, now this is done very seldom, just after trying to assign one hex connected to a one-time company ability, but I have now added game.is_loading.